iceoryx_hoofs/test/moduletests/test_memory_bump_allocator.cpp

```diff
@@ -0,0 +1,146 @@
+
+    void* memory{nullptr};
+    size_t memorySize = 10016;
+};
+
+TEST_F(BumpAllocator_Test, allocateOneSmallElement)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "f689e95c-5743-4370-93f0-8a23b909c75a");
+    iox::BumpAllocator sut(memory, memorySize);
+    auto allocationResult = sut.allocate(sizeof(int), MEMORY_ALIGNMENT);
+    ASSERT_FALSE(allocationResult.has_error());
+    int* bla = static_cast<int*>(allocationResult.value());
+    *bla = 123;
+    EXPECT_THAT(*bla, Eq(123));
+}
+
+TEST_F(BumpAllocator_Test, allocateEverythingWithSingleElement)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "f2e1085b-08fe-4b08-b022-0385b5a53fca");
+    iox::BumpAllocator sut(memory, memorySize);
+    auto allocationResult = sut.allocate(memorySize, 1);
+    ASSERT_FALSE(allocationResult.has_error());
+    int* bla = static_cast<int*>(allocationResult.value());
+    *bla = 123;
+    EXPECT_THAT(*bla, Eq(123));
+}
+
+TEST_F(BumpAllocator_Test, allocateEverythingWithMultipleElements)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "21d0fa61-54f9-41a0-8e53-e3448784497b");
+    iox::BumpAllocator sut(memory, memorySize);
+    for (size_t i = 0; i < memorySize; i += 32)
+    {
+        auto allocationResult = sut.allocate(32, 1);
+        ASSERT_FALSE(allocationResult.has_error());
+        auto* bla = static_cast<size_t*>(allocationResult.value());
+        *bla = i;
+        EXPECT_THAT(*bla, Eq(i));
+    }
+}
+
+TEST_F(BumpAllocator_Test, allocateTooMuchSingleElement)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "9deed5c0-19d8-4469-a5c3-f185d4d881f1");
+    iox::BumpAllocator sut(memory, memorySize);
+    auto allocationResult = sut.allocate(memorySize + 1, MEMORY_ALIGNMENT);
+    ASSERT_TRUE(allocationResult.has_error());
+    EXPECT_THAT(allocationResult.get_error(), Eq(iox::BumpAllocatorError::OUT_OF_MEMORY));
+}
+
+TEST_F(BumpAllocator_Test, allocateTooMuchMultipleElement)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "435151e8-cc34-41ce-8115-5c179716a60a");
+    iox::BumpAllocator sut(memory, memorySize);
+    for (size_t i = 0; i < memorySize; i += 32)
+    {
+        ASSERT_FALSE(sut.allocate(32, 1).has_error());
+    }
+
+    auto allocationResult = sut.allocate(1, MEMORY_ALIGNMENT);
+    ASSERT_TRUE(allocationResult.has_error());
+    EXPECT_THAT(allocationResult.get_error(), Eq(iox::BumpAllocatorError::OUT_OF_MEMORY));
+}
+
+TEST_F(BumpAllocator_Test, allocateAndAlignment)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "4252ddcc-05d4-499f-ad7c-30bffb420e08");
+    iox::BumpAllocator sut(memory, memorySize);
+    auto allocationResult = sut.allocate(5, MEMORY_ALIGNMENT);
+    ASSERT_FALSE(allocationResult.has_error());
+    auto* bla = static_cast<uint8_t*>(allocationResult.value());
+
+    allocationResult = sut.allocate(5, MEMORY_ALIGNMENT);
+    ASSERT_FALSE(allocationResult.has_error());
+    auto* bla2 = static_cast<uint8_t*>(allocationResult.value());
+    EXPECT_THAT(bla2 - bla, Eq(8U));
```

Comment:
That only shows that consecutive allocations are 8 bytes offset from each other but not that they are correctly aligned (more precisely it also assumes `bla2 > bla` which is OK for a bump allocator but not necessarily for others). Theoretically this condition does not need to hold, the gap could be larger and both addresses still correctly aligned. In this case some of the bytes are not used, but allocators do this all the time, depending on type.

The test should be changed to test the alignment of the addresses, meaning that they are multiples of whatever alignment is requested. The test name would be `allocationIsCorrectlyAligned`.